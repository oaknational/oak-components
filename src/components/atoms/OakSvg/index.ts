export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
