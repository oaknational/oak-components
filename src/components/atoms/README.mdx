# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
