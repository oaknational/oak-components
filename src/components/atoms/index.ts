export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
