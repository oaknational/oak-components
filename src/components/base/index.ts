export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
