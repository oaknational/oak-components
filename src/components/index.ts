export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
