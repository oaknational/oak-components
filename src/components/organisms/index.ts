export * from "./teacher";
export * from "./pupil";
export * from "./curriculum";
export * from "./shared";
