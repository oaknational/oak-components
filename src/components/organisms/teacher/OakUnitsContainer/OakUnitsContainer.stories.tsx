import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakUnitListItem } from "../OakUnitListItem";
import { OakUnitListOptionalityItem } from "../OakUnitListOptionalityItem";

import { OakUnitsContainer } from "./OakUnitsContainer";

import { OakInlineBanner } from "@/components";

const meta: Meta<typeof OakUnitsContainer> = {
  component: OakUnitsContainer,
  tags: ["autodocs"],
  argTypes: {
    isLegacy: { type: "boolean" },
    showHeader: { type: "boolean" },
    subject: { type: "string" },
    phase: { type: "string" },
    curriculumHref: {
      control: {
        type: "radio",
      },
      options: ["Url", "Null"],
      mapping: { Url: "https://www.thenational.academy", Null: null },
    },
    isCustomUnit: { type: "boolean" },
    customHeadingText: { type: "string" },
    banner: {
      control: {
        type: "select",
      },
      options: ["show", "hide"],
      mapping: {
        empty: [],
        show: (
          <OakInlineBanner
            isOpen={true}
            message={"Example banner text"}
            type="neutral"
            $width={"100%"}
          />
        ),
        hide: null,
      },
    },
  },
  parameters: {
    controls: {
      include: [
        "isLegacy",
        "showHeader",
        "subject",
        "phase",
        "curriculumHref",
        "isCustomUnit",
        "customHeadingText",
        "banner",
      ],
    },
  },
};

export default meta;

type Story = StoryObj<typeof OakUnitsContainer>;

export const Default: Story = {
  render: (args) => <OakUnitsContainer {...args} />,
  args: {
    isLegacy: false,
    showHeader: true,
    unitCards: [
      <OakUnitListItem
        index={1}
        title={"Lesson 1"}
        lessonCount={"4 lessons"}
        isLegacy={false}
        yearTitle={"Year 10"}
        href={"#"}
      />,

      <OakUnitListOptionalityItem
        nullTitle={
          "Numerals 1-10 this is a very long title (unavailable example) "
        }
        index={2}
        yearTitle="Year 11"
        optionalityUnits={[
          {
            title:
              "Migration: What do sources tell us about the British Empire in India and Africa?",

            href: "#",
            lessonCount: "10 lessons",
            slug: "migration",
          },
          {
            title:
              "Migration: What do sources tell us about the British Empire in India and Africa?",

            href: "#",
            lessonCount: "10 lessons",
            slug: "migration-2",
          },
          {
            title:
              "Migration: What do sources tell us about the British Empire in India and Africa?",

            href: "#",
            lessonCount: "10 lessons",
            slug: "migration-3",
          },
        ]}
        firstItemRef={undefined}
      />,
      <OakUnitListItem
        index={3}
        yearTitle={"Year 11"}
        title={"Lesson 3"}
        lessonCount={"2 lessons"}
        isLegacy={false}
        href={"#"}
      />,
      <OakUnitListItem
        index={4}
        title={
          "Lesson 4 - English  - a lesson with a longer title, a lesson with a longer title"
        }
        yearTitle={"Year 11"}
        lessonCount={"3 lessons"}
        isLegacy={false}
        href={"#"}
      />,
      <OakUnitListItem
        index={4}
        title={
          "Lesson 4 - English  - a lesson with a longer title, a lesson with a longer title"
        }
        yearTitle={"Year 11"}
        lessonCount={"1/3 lessons"}
        isLegacy={false}
        href={"#"}
      />,
    ],
    subject: "maths",
    phase: "secondary",
    curriculumHref: "Url",
  },
};
