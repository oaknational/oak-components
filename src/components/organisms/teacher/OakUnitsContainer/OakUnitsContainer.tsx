import React from "react";
import styled, { css } from "styled-components";

import {
  OakUnitsHeader,
  OakUnitsHeaderProps,
} from "@/components/organisms/teacher/OakUnitsHeader";
import { OakFlex, OakUL } from "@/components/atoms";
import { parseSpacing } from "@/styles/helpers/parseSpacing";

const OakULFlex = styled(OakUL)`
  display: flex;
  flex-direction: column;
  gap: ${parseSpacing("space-between-xs")};
`;

export type OakUnitsContainerProps = OakUnitsHeaderProps & {
  showHeader: boolean;
  unitCards: Array<React.ReactNode>;
  isCustomUnit?: boolean;
  customHeadingText?: string;
  bannerText?: string;
};

const OakUnitsContainerCss = css<OakUnitsContainerProps>``;

const UnstyledComponent = (props: OakUnitsContainerProps) => {
  const {
    isLegacy,
    showHeader,
    unitCards,
    curriculumHref,
    phase,
    subject,
    isCustomUnit,
    customHeadingText,
    bannerText,
    ...rest
  } = props;
  return (
    <OakFlex
      $gap="space-between-s"
      $alignItems="center"
      {...rest}
      $background={isLegacy ? "bg-neutral" : "bg-decorative3-very-subdued"}
      $flexDirection="column"
      $pa="inner-padding-m"
      $borderRadius="border-radius-m"
    >
      {showHeader && (
        <OakUnitsHeader
          isLegacy={isLegacy}
          curriculumHref={curriculumHref}
          phase={phase}
          subject={subject}
          $width="100%"
          isCustomUnit={isCustomUnit}
          customHeadingText={customHeadingText}
          bannerText={bannerText}
        />
      )}
      <OakULFlex aria-label="A list of units" $reset $width="100%">
        {...unitCards}
      </OakULFlex>
    </OakFlex>
  );
};

export const OakUnitsContainer = styled(UnstyledComponent)`
  ${OakUnitsContainerCss}
`;
