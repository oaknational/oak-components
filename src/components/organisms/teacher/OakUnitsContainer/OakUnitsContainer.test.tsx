import React from "react";
import "@testing-library/jest-dom";

import { OakUnitsContainer } from "./OakUnitsContainer";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakBox } from "@/components/atoms";

const props = {
  isLegacy: false,
  showHeader: true,
  unitCards: [
    <OakBox>Unit 1 </OakBox>,
    <OakBox>Unit 2 </OakBox>,
    <OakBox>Unit 3 </OakBox>,
    <OakBox>Unit 4 </OakBox>,
  ],
  subject: "maths",
  phase: "secondary",
  curriculumHref: "https://www.thenational.academy",
};

describe("OakUnitsContainer", () => {
  it("renders", () => {
    const { getByTestId } = renderWithTheme(
      <OakUnitsContainer data-testid="test" {...props} />,
    );
    expect(getByTestId("test")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakUnitsContainer {...props}></OakUnitsContainer>,
    );
    expect(container).toMatchSnapshot();
  });
});
