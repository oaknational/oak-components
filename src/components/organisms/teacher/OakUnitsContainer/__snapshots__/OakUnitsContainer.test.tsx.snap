// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakUnitsContainer matches snapshot 1`] = `
.c0 {
  padding: 1rem;
  background: #e3e9fb;
  border-radius: 0.375rem;
  font-family: Lexend,sans-serif;
}

.c2 {
  font-family: Lexend,sans-serif;
}

.c8 {
  position: relative;
  width: 2.5rem;
  font-family: Lexend,sans-serif;
}

.c12 {
  position: absolute;
  inset: 0rem;
  font-family: Lexend,sans-serif;
}

.c14 {
  font-family: Lexend,sans-serif;
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c16 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: Lexend,sans-serif;
}

.c23 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  color: #222222;
  background: #222222;
  border-radius: 6.25rem;
  font-family: Lexend,sans-serif;
}

.c24 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 6.25rem;
  font-family: Lexend,sans-serif;
}

.c25 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 1.5rem;
  min-height: 1.5rem;
  font-family: Lexend,sans-serif;
}

.c26 {
  -webkit-filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  filter: invert(98%) sepia(98%) saturate(0%) hue-rotate(328deg) brightness(102%) contrast(102%);
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 0.25rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.75rem;
}

.c7 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 2.5rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c13 {
  color: #ffe555;
  font-family: Lexend,sans-serif;
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c22 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c15 {
  font-family: Lexend,sans-serif;
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c27 {
  list-style: none;
  padding: 0;
  margin: 0;
  width: 100%;
  font-family: Lexend,sans-serif;
}

.c11 {
  fill: #222222;
}

.c19 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: Lexend,sans-serif;
  color: #222222;
  padding-top: 0.5rem;
}

.c19:disabled {
  pointer-events: none;
  cursor: default;
}

.c20 {
  display: inline-block;
  position: relative;
}

.c20:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #575757;
}

.c20:active {
  color: #222222;
}

.c20:disabled {
  color: #808080;
}

.c18 > :first-child:focus-visible .shadow {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c18 > :first-child:hover .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c18 > :first-child:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c18 > :first-child:disabled .icon-container {
  background: #808080;
}

.c18 > :first-child:hover .icon-container {
  background: #575757;
}

.c18 > :first-child:active .icon-container {
  background: #222222;
}

.c10 {
  aspect-ratio: 2 / 1;
}

.c4 {
  width: 100%;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.75rem;
}

@media (min-width:750px) {
  .c8 {
    width: 3rem;
  }
}

@media (min-width:1280px) {
  .c8 {
    width: 3.5rem;
  }
}

@media (min-width:750px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:750px) {
  .c3 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:750px) {
  .c13 {
    font-weight: 700;
  }
}

@media (min-width:1280px) {
  .c13 {
    font-weight: 600;
  }
}

@media (min-width:750px) {
  .c13 {
    font-size: 0.875rem;
  }
}

@media (min-width:1280px) {
  .c13 {
    font-size: 1rem;
  }
}

@media (min-width:750px) {
  .c13 {
    line-height: 1.25rem;
  }
}

@media (min-width:1280px) {
  .c13 {
    line-height: 1.25rem;
  }
}

@media (min-width:750px) {
  .c13 {
    -webkit-letter-spacing: -0.005rem;
    -moz-letter-spacing: -0.005rem;
    -ms-letter-spacing: -0.005rem;
    letter-spacing: -0.005rem;
  }
}

@media (min-width:1280px) {
  .c13 {
    -webkit-letter-spacing: 0.0115rem;
    -moz-letter-spacing: 0.0115rem;
    -ms-letter-spacing: 0.0115rem;
    letter-spacing: 0.0115rem;
  }
}

@media (min-width:750px) {
  .c19 {
    padding-top: 0rem;
  }
}

<div
  className="c0 c1 "
>
  <div
    className="c2 c3 c4"
  >
    <div
      className="c2 c5"
    >
      <div
        className="c2 c6"
      >
        <h4
          className="c7"
        >
          Maths units
        </h4>
        <div
          className="c8 c9 c10"
        >
          <svg
            className="c11"
            height="100%"
            viewBox="0 0 56 27"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.421 0c8.661.839 17.303.834 26.16.714a89.843 89.843 0 0 1 4.834-.013c6.333.32 12.39.317 18.506-.275 0 1.26.178 2.352 0 3.44-.276 1.38-.829 2.759-1.342 4.15-.138.354-.532.706-.808 1.06-.445.248-.63.5-.553.753 1.598 1.093-.256 2.185 0 3.277.276 1.529 1.322 3.058 1.204 4.572-.138 1.771-1.046 3.543-1.973 5.312-.288.4-1.065.792-2.309 1.169-2.288.755-4.103 1.67-11.206 1.965-9.936.417-20.953.385-30.678-.09-2.447-.12-4.38-.532-5.209-.853-1.035-.458-1.52-.929-1.44-1.4C.937 18.988-.109 14.194.01 9.4c0-2.645 1.973-5.29 3.413-7.933.237-.424 1.697-.839 3-1.468Z"
            />
          </svg>
          <div
            className="c12 c9"
          >
            <span
              className="c13"
            >
              New
            </span>
          </div>
        </div>
      </div>
      <div
        className="c14 c15"
      >
        Brand-new teaching resources, thoughtfully crafted by teachers for classroom needs.
      </div>
    </div>
    <div
      className="c16 c17 c18"
    >
      <a
        className="c19 c20"
        href="https://www.thenational.academy"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="c2 c21"
        >
          <span
            className="c22"
          >
            Full secondary curriculum
          </span>
          <div
            className="c23 c9 icon-container"
          >
            <div
              className="c24 shadow"
            />
            <div
              className="c25"
              data-icon-for="button"
            >
              <img
                alt="chevron-right"
                className="c26"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1707752509/icons/vk9xxxhnsltsickom6q9.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <ul
    aria-label="A list of units"
    className="c27 c28"
  >
    <div
      className="c2"
    >
      Unit 1 
    </div>
    <div
      className="c2"
    >
      Unit 2 
    </div>
    <div
      className="c2"
    >
      Unit 3 
    </div>
    <div
      className="c2"
    >
      Unit 4 
    </div>
  </ul>
</div>
`;
